### PR TITLE
Attributes minimization

### PR DESCRIPTION
**Pull request by ghost**
Added extra minimizable attributes and fixed the bug that those
attributes were not miminized.

### CODE REVIEW

**Comment by dereuromark** [comment]
Resolves https://github.com/cakephp/cakephp/issues/6560

File: src/View/StringTemplate.php [hunk 1]
```diff
@@ -284,7 +306,7 @@ protected function _formatAttribute($key, $value, $escape = true)
         $truthy = [1, '1', true, 'true', $key];
         $isMinimized = isset($this->_compactAttributes[$key]);
         if ($isMinimized && in_array($value, $truthy, true)) {
-            return "$key=\"$key\"";
+            return "$key";
```
Comment (dereuromark) [inline src/View/StringTemplate.php]: Looks good so far. But why this is necessary? As far as I know both formats are valid (both short and long syntax).
And this makes a lot of other tests fails, as well.

Comment (NaranjaWeb) [inline src/View/StringTemplate.php]: Yes, you are right. Not necessary very much.

I didn't know those two formats are the same and it took me long
how I could implement the shorten version of attributes in cake.

I found that the doc comments in the source say that you can make 
it compact, and this is another reason why I report it.

Now I know this is not a necessary change at all. Thank you for your care.

Comment (dereuromark) [inline src/View/StringTemplate.php]: Well, I don't mind either way :) Just saying that you will need to fix a lot of more tests for the short version.

Comment (markstory) [inline src/View/StringTemplate.php]: Short form attributes also break in xhtml doc types. This change would make using the helpers impossible in xhtml doc types.

**Comment by jhli** [comment]
I think default should always be xml compatible!

**Comment by NaranjaWeb** [comment]
I agree. Dismiss, or just adding the new attributes could be a fair conclusion.

**Comment by dereuromark** [comment]
@NaranjaWeb Maybe you can revise the PR regarding the comments and fix the tests accordingly.

File: src/View/StringTemplate.php [hunk 2]
```diff
@@ -37,26 +37,48 @@ class StringTemplate
...
-        'formnovalidate' => true,
+        'reversed' => true,
+        'scoped' => true,
+        'seamless' => true,
+        'selected' => true,
+        'sortable' => true,
+        'spellcheck' => true,
+        'truespeed' => true,
+        'typemustmatch' => true,
+        'visible ' => true,
```
Comment (dereuromark) [inline src/View/StringTemplate.php]: whitespace that should not be there in `'visible '`

Comment (josegonzalez) [inline src/View/StringTemplate.php]: It's invisible!

Comment (dereuromark) [inline src/View/StringTemplate.php]: :trollface:

**Comment by NaranjaWeb** [comment]
I revised PR, including the tests and a visible space. Thank you.

**Comment by dereuromark** [comment]
:+1: 
I backported it to 2.x.

**Comment by lorenzo** [comment]
Thanks!

**Pull request merged**